package/base-files/files/lib/upgrade/stage2

```diff
@@ -123,15 +123,27 @@ kill_remaining() { # [ <signal> [ <loop> ] ]
 
 indicate_upgrade
 
+skip_services="dnsmasq log network"
+
+for service in $(ls /etc/init.d); do
```

Comment:
I'd suggest to loop it like this:

```bash
for service in /etc/init.d/*; do
  service=${service##*/}
  ...
done
```

Advantage is that you do not need to spawn an extra subshell to call `ls`